File: java/ABTestExample/app/src/main/java/com/pubscale/sdkone/example/abtestexample/utils/AppOpenAdManager.java
package com.pubscale.sdkone.example.abtestexample.utils;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;
import androidx.lifecycle.ProcessLifecycleOwner;

import com.pubscale.sdkone.example.abtestexample.ad_formats.AppOpen;
import com.pubscale.sdkone.example.abtestexample.event_listener.AppOpenAdEventListener;

public class AppOpenAdManager implements Application.ActivityLifecycleCallbacks, LifecycleObserver {

    private Activity currentActivity = null;
    private boolean appStarting = true;
    private AppOpen appOpen = AppOpen.getInstance();
    private AppOpenAdEventListener appOpenAdEventListener = new AppOpenAdEventListener() {
        @Override
        public void onAdLoading() {

        }

        @Override
        public void onAdLoaded() {
            if(currentActivity != null) {
                appOpen.showAd(currentActivity, true);
            }
        }

        @Override
        public void onAdLoadFailed() {
        }

        @Override
        public void onAdShowFailed() {
        }

        @Override
        public void onAdOpened() {
        }

        @Override
        public void onAdClosed() {
        }
    };

    public AppOpenAdManager() {
        ProcessLifecycleOwner.get().getLifecycle().addObserver(this);
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    void onMoveToForeground() {
        if(appStarting) {
            appStarting = false;
            return;
        }
        if(currentActivity == null) return;
        appOpen.setAppOpenAdEventListener(appOpenAdEventListener).loadAd(currentActivity, true);
    }

    @Override
    public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState) {
    }

    @Override
    public void onActivityStarted(@NonNull Activity activity) {
        if(!appOpen.isShowingAd()) currentActivity = activity;
    }

    @Override
    public void onActivityResumed(@NonNull Activity activity) {
    }

    @Override
    public void onActivityPaused(@NonNull Activity activity) {
    }

    @Override
    public void onActivityStopped(@NonNull Activity activity) {
    }

    @Override
    public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {
    }

    @Override
    public void onActivityDestroyed(@NonNull Activity activity) {
    }
}

File: kotlin/ABTestExample/app/src/main/java/com/pubscale/sdkone/example/abtestexample/utils/AppOpenAdManager.kt
package com.pubscale.sdkone.example.abtestexample.utils

import android.app.Activity
import android.app.Application
import android.os.Bundle
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import androidx.lifecycle.ProcessLifecycleOwner
import com.pubscale.sdkone.example.abtestexample.ad_formats.AppOpen
import com.pubscale.sdkone.example.abtestexample.event_listener.AppOpenAdEventListener

class AppOpenAdManager : Application.ActivityLifecycleCallbacks, LifecycleObserver {

    private var currentActivity: Activity? = null
    private var appStarting = true
    private val appOpenAdEventListener = object : AppOpenAdEventListener {
        override fun onAdLoading() {
        }

        override fun onAdLoaded() {
            currentActivity?.let{ AppOpen.showAd(it, false) }
        }

        override fun onAdLoadFailed() {
        }

        override fun onAdShowFailed() {
        }

        override fun onAdOpened() {
        }

        override fun onAdClosed() {
        }

    }

    init {
        ProcessLifecycleOwner.get().lifecycle.addObserver(this)
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun onMoveToForeground() {
        if(appStarting) {
            appStarting = false
            return
        }
        currentActivity?.let {
            AppOpen.setAppOpenAdEventListener(appOpenAdEventListener).loadAd(it, false)
        }
    }

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
    }

    override fun onActivityStarted(activity: Activity) {
        if(!AppOpen.isShowingAd) currentActivity = activity
    }

    override fun onActivityResumed(activity: Activity) {
    }

    override fun onActivityPaused(activity: Activity) {
    }

    override fun onActivityStopped(activity: Activity) {
    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
    }

    override fun onActivityDestroyed(activity: Activity) {
    }
}